### PR TITLE
determine which semantic domains are actually used show only these in search dropdown

### LINKED ISSUE #501: Search by Semantic Domains should display in the dropdown box only those used in the project
When searching by Semantic Domains, the dropdown box should display only the semantic domains that are referenced in the database. Instead, it displays the entire list of semantic domains. (Only 4 different semantic domains are referenced in this database: 4.3.4.6.1, 5.2.1.4, 5.2.1.5, and 6.)
https://www.webonary.org/phuien?s=a&search=Search&key=pug&pos=&semantic_domain=5.2.1.5&search_options_set=1&lang=en
image

### PR DESCRIPTION
**Pull request by hanatgit**

### CODE REVIEW

File: webonary-cloud-api/lambda/getDictionary.ts [hunk 1]
```diff
@@ -81,9 +83,19 @@ export async function handler(event: APIGatewayEvent): Promise<APIGatewayProxyRe
   }
 
   // TODO: Populate these during dictionary upload
+  const entriesCollection = db.collection(dbCollectionEntries(dictionaryId));
+
+  // get total entries and semantic domains used in senses
+  [dbItem.mainLanguage.entriesCount, dbItem.semanticDomainAbbreviationsUsed] = await Promise.all([
```
Comment (johntseng) [inline webonary-cloud-api/lambda/getDictionary.ts:89]: Are these queries expensive? Maybe consider storing this in the dictionary entry during upload?

**Review by johntseng — APPROVED** [review]

File: webonary-cloud-api/lambda/getDictionary.ts (hunk 1 above)
Comment (hanatgit) [reply webonary-cloud-api/lambda/getDictionary.ts:89]: Not too expensive. And dictionary info is cached in Webonary until a new dictionary is uploaded. If we store them during dictionary upload, we still need to do this here, for legacy dictionaries, unless we do a data migration.

Once this project becomes stable, I think that would be a good time to do this logic during upload

File: wp-resources/plugins/sil-dictionary-webonary/webonary/Webonary_SemanticDomains.php [hunk 2]
```diff
@@ -379,10 +379,36 @@ public static function GetTranslatedList(string $lang_code = '', bool $include_d
 
 	public static function GetDropdown(string $lang_code = ''): string
 	{
-		// get the list of domains
-		$domains = self::GetTranslatedList($lang_code, false);
-		if (empty($domains))
-			return '';
+		// check for cached list
```
Comment (phillip-hopper) [inline wp-resources/plugins/sil-dictionary-webonary/webonary/Webonary_SemanticDomains.php:382]: @hanatgit, I added code to cache the filtered list.

**Review by phillip-hopper — APPROVED** [review]

File: wp-resources/plugins/sil-dictionary-webonary/webonary/Webonary_SemanticDomains.php (hunk 2 above)
Comment (hanatgit) [reply wp-resources/plugins/sil-dictionary-webonary/webonary/Webonary_SemanticDomains.php:382]: Great idea, @phillip-hopper ! Your code LGTM.

**Pull request merged**